lib/hets/prove/evaluator.rb

```diff
@@ -13,17 +13,19 @@ class Evaluator
       # and
       # new(user, ontology, path: some_path, io: some_io)
       # where some_io needs to be an instance of IO or a Tempfile.
-      def initialize(user, ontology, version: nil, path: nil, io: nil)
+      def initialize(user, ontology, proof_attempts,
+                     version: nil, path: nil, io: nil)
         self.version = version
         self.path = path
         self.ontology = ontology
+        self.proof_attempts = proof_attempts
         self.user = user
         self.io = io
         initialize_handling
       end
 
       def import
-        callback = ProveEvaluator.new(self)
+        callback = ProveEvaluator.new(self, proof_attempts)
         ActiveRecord::Base.transaction requires_new: true do
           callback.process(:all, :start)
           parser.parse(callback: callback)
```

Comment:
It is hard for me to get the overall structure here. Why is there lib/hets/prove/evaluator.rb and lib/hets/prove/prove_evaluator.rb? Why does ProveEvaluator inherit from BaseEvaluator, while Evaluator does inherit anything (and seems to be unrelated to the other two)?